gprofiler/utils/node.py

```diff
@@ -0,0 +1,223 @@
+from functools import lru_cache
+import shutil
+import traceback
+from typing import Callable
+from gprofiler.metadata import get_exe_version
+from gprofiler.log import get_logger_adapter
+from granulate_utils.linux.ns import get_proc_root_path, is_same_ns, NsType, run_in_ns
+from granulate_utils.linux.process import is_musl
+
+
+from . import add_permission_dir, pgrep_maps, resource_path, pgrep_exe
+
+import json
+import requests
+import os
+import signal
+import psutil
+import stat
+
+from retry import retry
+from websocket import create_connection
+from websocket._core import WebSocket
+
+logger = get_logger_adapter(__name__)
+
+
+class NodeDebuggerUrlNotFound(Exception):
+    pass
+
+
+class NodeDebuggerUnexpectedResponse(Exception):
+    pass
+
+
+def _start_debugger(pid):
+    os.kill(pid, signal.SIGUSR1)
+
+
+def get_node_processes():
+    return pgrep_maps(r"(?:^.+/node[^/]*$)")
```

Comment:
I think for `node` you could just use `pgrep_exe("node")`. The idea of `pgrep_maps` is for Java/Python where the exe is not necessarily `java` or `python` but can be e.g `uwsgi` and load `libpython.so`. In node - it's always `node` binary.

Reply:
I'm a little bit worried about nexe / electron, but I don't think that they are common use cases and I don't even know if this method of generating symbols will work for them, so I think we can use `pgrep_exe`

Reply:
I don't know if they're used in production, I have never seen them used so I think we can skip them for now.